### PR TITLE
patch registry fix of issue 2553

### PR DESCRIPTION
**Pull request by wy65701436**
This commit is target to fix harbor issue #9186, which root cause is mentioned by
docker/distribution#2553, and fixed by docker/distribution#2879.

As the latest distribution release(v2.7.1) does not contain this fix, but it will break the quota migraion process on S3 storage, we have to path this fix into Harbor regsitry binary.

[Tag Version]
It uses the issue number(2553) as the tag naming convention, like v2.7.1-patch-2553, means that we patch the fix of issue 2553 into v2.7.1.

[Note]
So far, this fix is only targets on docker regsitry v2.7.1. If the registry has this fix in new release, we'll move on.

Signed-off-by: wang yan <wangyan@vmware.com>

### CODE REVIEW

File: make/photon/registry/builder [hunk 1]
```diff
@@ -22,6 +22,13 @@ cur=$PWD
 TEMP=`mktemp -d /$TMPDIR/distribution.XXXXXX`
 git clone -b $VERSION https://github.com/docker/distribution.git $TEMP
 
+# add patch 2879
```
Comment (renmaosheng) [inline make/photon/registry/builder]: is it 2553 or 2879? it seems difference from the REGISTRYVERSION above.

Comment (wy65701436) [reply make/photon/registry/builder]: 2879 is the pull request number,
2553 is the github issue number, we will the issue number as tag, it's better for user to get more detail and background.

Comment (reasonerjt) [reply make/photon/registry/builder]: Can you explain this in the comment?  Plus, let's also clarify this is only needed for registry 271

**Comment by reasonerjt** [comment]
The change in registry looks good to me, let's add some comment and merge this PR

File: make/photon/registry/builder (hunk 1 above)
Comment (wy65701436) [reply make/photon/registry/builder]: added in the commit-msg.

**Pull request merged**